pkgs/applications/networking/browsers/mullvad-browser/default.nix

```diff
@@ -1,233 +1,56 @@
 { lib
+, buildMozillaMach
+, fetchFromGitHub
 , stdenv
-, fetchurl
-, makeDesktopItem
-, copyDesktopItems
-, makeWrapper
-, writeText
-, wrapGAppsHook
-
-# Common run-time dependencies
-, zlib
-
-# libxul run-time dependencies
-, atk
-, cairo
-, dbus
-, dbus-glib
-, fontconfig
-, freetype
-, gdk-pixbuf
-, glib
-, gtk3
-, libxcb
-, libX11
-, libXext
-, libXrender
-, libXt
-, libXtst
-, mesa
-, pango
-, pciutils
-
-, libnotifySupport ? stdenv.isLinux
-, libnotify
-
-, audioSupport ? mediaSupport
-, pulseaudioSupport ? mediaSupport
-, libpulseaudio
-, apulse
-, alsa-lib
-
-# Media support (implies audio support)
-, mediaSupport ? true
-, ffmpeg
-
-# Extra preferences
-, extraPrefs ? ""
+, firefox-esr
+, nix-update-script
 }:
-
-let
-  libPath = lib.makeLibraryPath (
-    [
-      alsa-lib
-      atk
-      cairo
-      dbus
-      dbus-glib
-      fontconfig
-      freetype
-      gdk-pixbuf
-      glib
-      gtk3
-      libxcb
-      libX11
-      libXext
-      libXrender
-      libXt
-      libXtst
-      mesa # for libgbm
-      pango
-      pciutils
-      stdenv.cc.cc
-      stdenv.cc.libc
-      zlib
-    ] ++ lib.optionals libnotifySupport [ libnotify ]
-      ++ lib.optionals pulseaudioSupport [ libpulseaudio ]
-      ++ lib.optionals mediaSupport [ ffmpeg ]
-  );
-
-  version = "12.0.6";
-
-  srcs = {
-    x86_64-linux = fetchurl {
-      url = "https://cdn.mullvad.net/browser/${version}/mullvad-browser-linux64-${version}_ALL.tar.xz";
-      hash = "sha256-XE6HFU38FhnikxGHRHxIGS3Z3Y2JNWH0yq2NejqbROI=";
-    };
+((buildMozillaMach rec {
+  pname = "mullvad-browser";
+  applicationName = "Mullvad Browser";
+  binaryName = "mullvadbrowser";
+  branding = "browser/branding/mb-release";
+  # mullvad-browser is based on firefox ESR but since we can't properly keep track of the exact ESR version it uses,
+  # we will just inherit it here from the version in nixpkgs. The reason we need to use a firefox version is because
+  # some patches are applied conditionally with lib.versionAtLeast.
+  version = "${firefox-esr.version}-dummy"; # TODO: find a better solution as this will trigger rebuilds if changed.
```

Comment:
I think we just have to set the actual ESR version here manually, even if it's annoying to update.